appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java

```diff
@@ -1599,16 +1602,18 @@ protected void initializeHome()
             JndiInfo jndiInfo = entry.getValue();
             try {
                 jndiInfo.publish(this.namingManager);
-                if ( jndiInfo.internal ) {
+                if (jndiInfo.internal) {
                     publishedInternalGlobalJndiNames.add(jndiInfo.name);
                 } else {
-                    if ( jndiInfo.portable ) {
+                    if (jndiInfo.portable) {
                         publishedPortableGlobalJndiNames.add(jndiInfo.name);
-                    }  else {
+                    } else {
                         publishedNonPortableGlobalJndiNames.add(jndiInfo.name);
                     }
                 }
-            } catch(Exception e) {
+            } catch (NameAlreadyBoundException e) {
```

Comment:
Not 100% sure about this, but seems okay.

Reply:
i found the real cause of the issue for the remoteview, a duplicated call for the initializeHome method